# Data Understanding & Initial Data Audit

### Import Required Libraries

In [1]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns 
import plotly.express as px


### Load Datatset

In [2]:
df = pd.read_excel("../data/raw/indian-job-market-dataset-2025.xlsx")

In [3]:
df.shape

(97929, 17)

### Column Names

In [4]:
df.columns

Index(['title', 'jobId', 'currency', 'jobUploaded', 'companyName',
       'tagsAndSkills', 'experience', 'salary', 'location', 'companyId',
       'ReviewsCount', 'AggregateRating', 'jobDescription', 'minimumSalary',
       'maximumSalary', 'minimumExperience', 'maximumExperience'],
      dtype='str')

In [5]:
df.head()

,title,jobId,currency,jobUploaded,companyName,tagsAndSkills,experience,salary,location,companyId,ReviewsCount,AggregateRating,jobDescription,minimumSalary,maximumSalary,minimumExperience,maximumExperience
0,Sr. HR Recruiter (NON IT),270925008041,INR,6 Days Ago,Orion,"Communication,Manpower,Staffing,Convincing Pow...",2-4 Yrs,2-4 Lacs PA,Kolkata(Chinar Park),645563,NaN,NaN,Preferred candidate profile . .,200000.0,400000.0,2.0,4.0
1,Fire And Safety Officer,270925007584,INR,6 Days Ago,"Apollo Hospitals International Limited, Ahmedabad","Safety Officer Activities,Fire Protection,Fire...",6-11 Yrs,3-5 Lacs PA,"Gandhinagar, Ahmedabad",14072,5162.0,4.0,"Ensure active Fire Protection System,such as F...",300000.0,500000.0,6.0,11.0
2,Opening For Performance Marketing - Chennai,270925007492,INR,6 Days Ago,TVS Credit Services Ltd,"Performance Marketing,User Acquisition,growth ...",12-18 Yrs,Not disclosed,Chennai,1324750,2892.0,4.2,MBA Marketing (preferred Tier II or III B- Sch...,0.0,0.0,12.0,18.0
3,Medical Billing Executive,270925007443,INR,6 Days Ago,GNR Global Services,"Fluent English,Spoken English,Good English Com...",0-3 Yrs,"70,000-2 Lacs PA","Mohali, Chandigarh, Kharar, Zirakpur",123804403,NaN,NaN,Job Title-Medical Billing Executive\nLocation-...,70000.0,200000.0,0.0,3.0
4,Senior Group Product Manager - CNS Therapy,270925007430,INR,6 Days Ago,Cadila Pharmaceuticals,"Product Marketing,CNS,Product Management,Nephr...",5-10 Yrs,8-18 Lacs PA,Ahmedabad,14957,2134.0,3.4,Principal Tasks & Responsibilities : (Please w...,800000.0,1800000.0,5.0,10.0


In [6]:
df.tail()

,title,jobId,currency,jobUploaded,companyName,tagsAndSkills,experience,salary,location,companyId,ReviewsCount,AggregateRating,jobDescription,minimumSalary,maximumSalary,minimumExperience,maximumExperience
97924,"CNC Machine Maintenance Engineer @ Manali, Che...",270925008790,INR,6 Days Ago,GSH Group,"CNC Maintenance,CNC Machines,Electrical Mainte...",2-5 Yrs,"50,000-2 Lacs PA","Chennai, Thiruvallur",938550,284.0,4.2,Experience: Minimum 2+ Years in CNC Machine El...,50000.0,200000.0,2.0,5.0
97925,IT Cum Digital Marketing Executive,270925008668,INR,6 Days Ago,Luxus Green Mobility Pvt. Ltd.,"Google AdWords,Digital Marketing,Social Media ...",1-2 Yrs,Not disclosed,Bilaspur,123904985,13.0,2.6,<p> </p><p><strong>Position Overview :-</stron...,0.0,0.0,1.0,2.0
97926,Senior Research Associate,270925008467,INR,6 Days Ago,Cadila Pharmaceuticals,"Analytical Method Validation,Analytical Method...",4-9 Yrs,Not disclosed,Ahmedabad,14957,2134.0,3.4,Documentation as per GLP requirement,0.0,0.0,4.0,9.0
97927,Customer Care Associate,270925008099,INR,6 Days Ago,GNR Global Services,"Fluent English,Spoken English,Strong Communica...",0-3 Yrs,"70,000-2 Lacs PA","Mohali, Chandigarh, Zirakpur",123804403,NaN,NaN,Job title-Customer Care Associate\nJob locatio...,70000.0,200000.0,0.0,3.0
97928,Client Relationship Executive,270925008094,INR,6 Days Ago,Logon Global Solutions,"Client Support,English,Client Coordination,Mal...",0-3 Yrs,1.92-3.36 Lacs PA,Kochi(Kakkanad),125008981,4.0,4.2,<p> <strong>Role:</strong> Primary advisor for...,192000.0,336000.0,0.0,3.0


### Dataset Information

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 97929 entries, 0 to 97928
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   title              97929 non-null  str    
 1   jobId              97929 non-null  int64  
 2   currency           97929 non-null  str    
 3   jobUploaded        97929 non-null  str    
 4   companyName        97925 non-null  str    
 5   tagsAndSkills      97358 non-null  str    
 6   experience         95824 non-null  str    
 7   salary             97929 non-null  str    
 8   location           97929 non-null  str    
 9   companyId          97929 non-null  int64  
 10  ReviewsCount       62677 non-null  float64
 11  AggregateRating    62677 non-null  float64
 12  jobDescription     97929 non-null  str    
 13  minimumSalary      97358 non-null  float64
 14  maximumSalary      97358 non-null  float64
 15  minimumExperience  97358 non-null  float64
 16  maximumExperience  97358 non-null

### Numerical Summary

In [8]:
df.describe()

,jobId,companyId,ReviewsCount,AggregateRating,minimumSalary,maximumSalary,minimumExperience,maximumExperience
count,9.792900e+04,9.792900e+04,62677.000000,62677.000000,9.735800e+04,9.735800e+04,97358.000000,97358.000000
mean,2.069854e+11,2.784642e+07,14662.590264,3.679048,1.983092e+05,3.186096e+05,3.957744,7.732647
std,9.987860e+10,4.979184e+07,24744.514682,0.484096,5.640765e+05,8.417336e+05,3.581196,4.500161
min,1.012500e+10,3.000000e+00,2.000000,1.000000,0.000000e+00,0.000000e+00,0.000000,0.000000
25%,1.808250e+11,2.339390e+05,65.000000,3.500000,0.000000e+00,0.000000e+00,1.000000,5.000000
50%,2.508250e+11,2.892790e+06,962.000000,3.700000,0.000000e+00,0.000000e+00,3.000000,7.000000
75%,2.808250e+11,6.398332e+06,12826.000000,3.900000,2.000000e+05,3.750000e+05,5.000000,10.000000
max,3.112240e+11,1.251422e+08,104449.000000,5.000000,7.000000e+07,9.500000e+07,30.000000,35.000000


### Categorical Summary

In [9]:
df.describe(include=["object", "string"])

,title,currency,jobUploaded,companyName,tagsAndSkills,experience,salary,location,jobDescription
count,97929,97929,97929,97925,97358,95824,97929,97929,97929
unique,55104,2,30,18668,84307,287,1339,10066,80679
top,Application Developer,INR,4 Days Ago,Accenture,"Retail,Lead generation,Forex,Investment produc...",5-10 Yrs,Not disclosed,Bengaluru,Disclaimer: This job description has been sour...
freq,2039,97800,13973,8204,116,6448,64076,16823,382


### Missing Values

In [10]:
df.isnull().sum()

title                    0
jobId                    0
currency                 0
jobUploaded              0
companyName              4
tagsAndSkills          571
experience            2105
salary                   0
location                 0
companyId                0
ReviewsCount         35252
AggregateRating      35252
jobDescription           0
minimumSalary          571
maximumSalary          571
minimumExperience      571
maximumExperience      571
dtype: int64

### Duplicate Values

In [11]:
df.duplicated()

0        False
1        False
2        False
3        False
4        False
         ...  
97924    False
97925    False
97926    False
97927    False
97928    False
Length: 97929, dtype: bool

### Data Types

In [12]:
df.dtypes

title                    str
jobId                  int64
currency                 str
jobUploaded              str
companyName              str
tagsAndSkills            str
experience               str
salary                   str
location                 str
companyId              int64
ReviewsCount         float64
AggregateRating      float64
jobDescription           str
minimumSalary        float64
maximumSalary        float64
minimumExperience    float64
maximumExperience    float64
dtype: object

### Unique Values

In [13]:
df["title"].unique()

<ArrowStringArray>
[                          'Sr. HR Recruiter (NON IT)',
                             'Fire And Safety Officer',
         'Opening For Performance Marketing - Chennai',
                           'Medical Billing Executive',
         'Senior Group Product Manager -  CNS Therapy',
                                   'Senior Accountant',
                              'Maintenance Technician',
                          'Officer/Engineer Technical',
                              'Officer- Technical Lab',
                           'Site Reliability Engineer',
 ...
              'Sales Executive-Food/ Horeca-Bangalore',
                        'hiring For international BPO',
                   'Remote Support Engineer - Chennai',
                           'Assistant Manager - Admin',
                'Graphic Designer (with UI/UX Skills)',
                    'Senior Executive - Fire & Safety',
                          'B2B Sales Manager - Mohali',
 'Sr.Executive/Asst. Man

In [14]:
df["jobId"].unique()

array([270925008041, 270925007584, 270925007492, ..., 270925008467,
       270925008099, 270925008094], shape=(97679,))

In [15]:
df["currency"].unique()

<ArrowStringArray>
['INR', 'USD']
Length: 2, dtype: str

In [16]:
df["jobUploaded"].unique()

<ArrowStringArray>
[           '6 Days Ago',         'Few Hours Ago',              'Just Now',
            '3 Days Ago',            '4 Days Ago',             '1 Day Ago',
            '2 Days Ago',            '7 Days Ago',            '5 Days Ago',
                 'Today',            '8 Days Ago',  'Starts in 1-3 months',
            '9 Days Ago',           '10 Days Ago', 'Starts within 1 month',
  'Starts : 6th Oct' 25',  'Starts : 3rd Nov' 25',  'Starts : 8th Oct' 25',
  'Starts in 3-6 months', 'Starts : 10th Oct' 25',  'Starts : 7th Oct' 25',
 'Starts : 20th Oct' 25', 'Starts : 22nd Oct' 25', 'Starts : 25th Nov' 25',
 'Starts : 15th Oct' 25', 'Starts : 15th Jan' 26', 'Starts : 14th Oct' 25',
           '15 Days Ago',           '20 Days Ago',  'Starts : 1st Nov' 25']
Length: 30, dtype: str

In [17]:
df["companyName"].unique()

<ArrowStringArray>
[                                            'Orion',
 'Apollo Hospitals International Limited, Ahmedabad',
                           'TVS Credit Services Ltd',
                               'GNR Global Services',
                            'Cadila Pharmaceuticals',
                                'Vedaant Techno Art',
                               'Avalon Technologies',
                       'Balkrishna Industries (BKT)',
                                           'Coforge',
                                      'Knorr Bremse',
 ...
                          'Bodysutra Healthtech Llp',
                                  'Vishal Mega Mart',
                                 'Atul C Doshi & Co',
            'Datamatics Global Services LTD, Nashik',
                                     'Ajanta Bottle',
                                      'Jasbir Singh',
                            'Green Rider Technology',
                                              'Ipca',
    

In [18]:
df["tagsAndSkills"].unique()

<ArrowStringArray>
[                                             'Communication,Manpower,Staffing,Convincing Power,Hiring,Recruitment,SR,Communication skills',
 'Safety Officer Activities,Fire Protection,Fire Safety,Fire Engineering,Fire Prevention,Safety Management,Fire Management,Hazard Analysis',
                           'Performance Marketing,User Acquisition,growth marketing,Paid Marketing,Acquisition,Performance,Usage,Marketing',
                                  'Fluent English,Spoken English,Good English Communication,Medical,Medical billing,Billing,Fluent,English',
                  'Product Marketing,CNS,Product Management,Nephrology,Group Product Management,Brand Marketing,Neurology,Brand Management',
                            'Tds Filing,ITR,Gst Filing,Balance Sheet Finalisation,Tax Audit,Return Filing,Accounts Finalisation,E Way Bill',
                               'Booth,Conveyor,Special Process,Utility Maintenance,Powder Coating,Preventive Maintenance,Paint Shop,Pow

In [19]:
df["experience"].unique()

<ArrowStringArray>
[  '2-4 Yrs',  '6-11 Yrs', '12-18 Yrs',   '0-3 Yrs',  '5-10 Yrs', '10-15 Yrs',
   '3-5 Yrs',   '2-3 Yrs',  '4-14 Yrs',   '0-4 Yrs',
 ...
 '12-21 Yrs',  '1-13 Yrs',  '0-12 Yrs',  '0-14 Yrs',  '3-17 Yrs',  '8-25 Yrs',
 '21-30 Yrs',    '13 Yrs',  '1-17 Yrs', '19-29 Yrs']
Length: 288, dtype: str

In [20]:
df["salary"].unique()

<ArrowStringArray>
[                            '2-4 Lacs PA',
                             '3-5 Lacs PA',
                           'Not disclosed',
                        '70,000-2 Lacs PA',
                            '8-18 Lacs PA',
                           '3.5-6 Lacs PA',
                     '50,000-1.75 Lacs PA',
                      '50,000-2.5 Lacs PA',
                           '3-4.5 Lacs PA',
                           '2-4.5 Lacs PA',
 ...
                        '96,000-3 Lacs PA',
                           '2-3.6 Lacs PA',
                        '3.24-4.8 Lacs PA',
                              '30 Lacs PA',
                           '40-95 Lacs PA',
 '5-10 Lacs PA (Including Variable: 5.0%)',
  '5-7 Lacs PA (Including Variable: 5.0%)',
                         '9-10.08 Lacs PA',
                        '36,000-48,000 PA',
                       '1.92-3.36 Lacs PA']
Length: 1339, dtype: str

In [21]:
df["location"].unique()

<ArrowStringArray>
[                            'Kolkata(Chinar Park)',
                           'Gandhinagar, Ahmedabad',
                                          'Chennai',
             'Mohali, Chandigarh, Kharar, Zirakpur',
                                        'Ahmedabad',
                             'Noida, Greater Noida',
              'Sriperumbudur, Chennai, Kanchipuram',
                                'Aurangabad(Waluj)',
               'New Delhi, Gurugram, Greater Noida',
                               'Mohali, Chandigarh',
 ...
                            'Lucknow, Meerut, Agra',
                          'Kochi, Kollam, Thrissur',
               'Faridabad, Delhi / NCR(Peeragarhi)',
       'Meerut, Moradabad, Delhi / NCR(Peeragarhi)',
                       'Jalandhar, Fazilka, Shimla',
                         'Jind, Moradabad, Bhiwani',
 'Ahmedabad(Ahmedabad Cantonment), Surat, Vadodara',
                             'Bengaluru(Hennur +2)',
                 'Ahme

In [22]:
df["companyId"].unique()

array([   645563,     14072,   1324750, ..., 125129245, 124078374,
       123904985], shape=(18328,))

In [23]:
df["ReviewsCount"].unique()

array([  nan, 5162., 2892., ...,  495.,  558., 3178.], shape=(1376,))

In [24]:
df["AggregateRating"].unique()

array([nan, 4. , 4.2, 3.4, 3.7, 4.1, 3.2, 3.5, 3.6, 3.9, 1.4, 3.8, 2.5,
       4.4, 3. , 4.3, 3.1, 2.9, 1.9, 5. , 2.4, 4.6, 3.3, 4.5, 1.3, 2.7,
       1.1, 2.3, 2.1, 4.9, 2.8, 2.6, 2. , 4.8, 4.7, 1.8, 1.6, 1.7, 1. ,
       1.2, 1.5, 2.2])

In [25]:
df["jobDescription"].unique()

<ArrowStringArray>
[                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    

In [26]:
df["minimumSalary"].unique()

array([2.000e+05, 3.000e+05, 0.000e+00, 7.000e+04, 8.000e+05, 3.500e+05,
       5.000e+04, 6.000e+05, 4.500e+05, 4.750e+05, 1.000e+06, 7.000e+05,
       4.000e+05, 1.500e+05, 3.250e+05, 2.000e+06, 9.000e+05, 3.750e+05,
       8.000e+04, 2.500e+05, 1.440e+05, 5.000e+05, 1.000e+05, 5.500e+05,
       3.000e+06, 1.400e+06, 2.160e+05,       nan, 6.000e+06, 3.500e+06,
       1.250e+05, 2.250e+05, 1.200e+05, 2.750e+05, 9.000e+04, 1.750e+05,
       4.250e+05, 1.200e+06, 6.500e+05, 7.500e+05, 4.000e+06, 9.500e+05,
       5.000e+06, 6.000e+04, 3.600e+05, 1.500e+06, 1.800e+06, 4.800e+05,
       1.800e+05, 3.360e+05, 2.640e+05, 2.400e+05, 9.600e+05, 2.500e+06,
       1.700e+06, 8.500e+05, 1.300e+06, 1.100e+06, 2.250e+06, 3.250e+06,
       2.500e+04, 3.480e+05, 1.560e+05, 1.600e+06, 9.600e+04, 1.900e+06,
       1.200e+04, 3.000e+04, 4.800e+04, 6.000e+03, 5.400e+05, 4.500e+06,
       1.680e+05, 3.600e+04, 3.960e+05, 3.840e+05, 2.400e+04, 4.200e+05,
       7.200e+05, 2.000e+07, 7.500e+06, 2.760e+05, 

In [27]:
df["maximumSalary"].unique()

array([4.000e+05, 5.000e+05, 0.000e+00, 2.000e+05, 1.800e+06, 6.000e+05,
       1.750e+05, 2.500e+05, 4.500e+05, 9.000e+05, 1.200e+06, 4.750e+05,
       1.300e+06, 1.100e+06, 1.500e+06, 3.000e+05, 6.500e+05, 3.500e+06,
       1.250e+05, 4.250e+05, 1.800e+05, 5.500e+05, 1.000e+06, 7.000e+05,
       5.000e+04, 1.550e+06, 4.500e+06, 2.400e+06, 3.500e+05, 3.600e+05,
             nan, 2.750e+05, 7.000e+04, 6.500e+06, 2.250e+05, 4.000e+06,
       3.750e+05, 2.400e+05, 1.500e+05, 5.000e+06, 8.000e+05, 3.250e+05,
       1.900e+06, 7.500e+05, 6.000e+06, 1.000e+07, 2.000e+06, 2.500e+06,
       9.600e+05, 7.200e+05, 4.080e+05, 4.200e+05, 1.400e+06, 3.000e+06,
       8.500e+05, 1.440e+06, 7.000e+06, 2.640e+05, 5.400e+05, 1.920e+05,
       2.250e+06, 9.500e+05, 1.600e+06, 1.700e+06, 8.750e+05, 2.300e+06,
       2.750e+06, 2.100e+06, 3.250e+06, 3.750e+06, 8.000e+04, 3.500e+04,
       7.750e+05, 1.680e+05, 3.300e+06, 4.800e+05, 9.600e+04, 9.000e+04,
       3.600e+04, 2.160e+05, 6.000e+04, 1.200e+05, 

In [28]:
df["minimumExperience"].unique()

array([ 2.,  6., 12.,  0.,  5., 10.,  3.,  4.,  7.,  8.,  1.,  9., 15.,
       nan, 13., 18., 14., 17., 11., 16., 20., 21., 22., 19., 24., 25.,
       23., 27., 30., 28., 29., 26.])

In [29]:
df["maximumExperience"].unique()

array([ 4., 11., 18.,  3., 10., 15.,  5., 14.,  8.,  6.,  7., 12., 13.,
        9., 19., 20.,  2., 16.,  1., nan,  0., 23., 24., 22., 25., 17.,
       28., 21., 31., 27., 30., 26., 29., 35.])

### Top Companies

In [30]:
df["companyName"].value_counts().head(20)

companyName
Accenture                                         8204
IDESLABS PRIVATE LIMITED                          2303
Wipro                                             1496
Krazy Mantra HR Solutions Pvt. Ltd                1028
Kotak Mahindra Bank                                987
PRO Hr Complete Solutions                          982
Infosys                                            938
Capgemini                                          700
IBM                                                687
PRADEEPIT CONSULTING SERVICES PVT LTD              618
Talent21                                           613
HR Innovation Group                                607
Vision Beyond Resources India Private Limited      582
EY                                                 571
Gratitude India Manpower Consultants Pvt. Ltd.     567
Axis Bank                                          547
ABB                                                428
Essentiallysports                                  42

### Top Locations

In [31]:
df["location"].value_counts().head(20)

location
Bengaluru                        16823
Hyderabad                         7669
Pune                              6112
Chennai                           4431
Gurugram                          4233
Mumbai                            4157
Noida                             2695
Remote                            1961
Ahmedabad                         1726
Navi Mumbai                       1453
Kolkata                           1049
Hyderabad, Chennai, Bengaluru     1037
Hybrid - Bengaluru                 946
Jaipur                             821
Coimbatore                         618
Hybrid - Hyderabad                 576
Kochi                              528
Vadodara                           528
Hybrid - Pune                      480
Thane                              410
Name: count, dtype: int64

### Top Ratings

In [32]:
df["AggregateRating"].value_counts().head(20)

AggregateRating
3.7    16684
3.9     5798
4.0     4529
3.5     4230
3.8     4071
3.4     3649
3.6     3451
4.1     3344
3.3     2045
3.2     1945
4.2     1898
3.0     1529
3.1     1508
4.3      863
4.4      753
4.8      699
2.9      631
4.5      627
5.0      581
2.8      578
Name: count, dtype: int64

### Leading/Trailing Spaces

In [33]:
space_counts = {
    col: (df[col].astype(str) != df[col].astype(str).str.strip()).sum()
    for col in df.select_dtypes(include=["object", "string"]).columns
}

pd.DataFrame(space_counts.items(), columns=["Column", "Leading/Trailing Spaces"])

,Column,Leading/Trailing Spaces
0,title,275
1,currency,0
2,jobUploaded,0
3,companyName,454
4,tagsAndSkills,586
5,experience,2105
6,salary,0
7,location,0
8,jobDescription,4554


### Empty Strings

In [34]:
empty_counts = {
    col: (df[col].astype(str).str.strip() == "").sum()
    for col in df.select_dtypes(include=["object", "string"]).columns
}

pd.DataFrame(empty_counts.items(), columns=["Column", "Empty Strings"])

,Column,Empty Strings
0,title,0
1,currency,0
2,jobUploaded,0
3,companyName,0
4,tagsAndSkills,0
5,experience,0
6,salary,0
7,location,0
8,jobDescription,0


### Minimum/ Maximum Values

In [35]:
df.min(numeric_only=True), df.max(numeric_only=True)

(jobId                1.012500e+10
 companyId            3.000000e+00
 ReviewsCount         2.000000e+00
 AggregateRating      1.000000e+00
 minimumSalary        0.000000e+00
 maximumSalary        0.000000e+00
 minimumExperience    0.000000e+00
 maximumExperience    0.000000e+00
 dtype: float64,
 jobId                3.112240e+11
 companyId            1.251422e+08
 ReviewsCount         1.044490e+05
 AggregateRating      5.000000e+00
 minimumSalary        7.000000e+07
 maximumSalary        9.500000e+07
 minimumExperience    3.000000e+01
 maximumExperience    3.500000e+01
 dtype: float64)

## Initial Data Audit Report

### 1. Dataset Overview
- The dataset contains **97,929 job postings** across **17 attributes**.
- It includes information related to job titles, companies, salaries, required experience, locations, skills, company ratings, reviews, and detailed job descriptions.
- The dataset appears comprehensive and is suitable for performing exploratory data analysis, visualization, dashboard development, SQL querying, and further feature engineering.

---

### 2. Data Structure
- The dataset contains a combination of numerical and categorical features.
- Numerical columns include salary ranges, experience ranges, ratings, and review counts.
- Categorical columns include job titles, company names, locations, uploaded dates, skills, and job descriptions.

---

### 3. Missing Values
- Missing values are present in certain columns, particularly **ReviewsCount** and **AggregateRating**.
- These missing values will be examined during the data cleaning phase to determine the most appropriate treatment strategy.

---

### 4. Duplicate Records
- Duplicate records were inspected.
- No duplicate rows were identified during the initial audit.

---

### 5. Data Quality
- Initial inspection was performed for:
  - Missing values
  - Duplicate records
  - Data types
  - Leading and trailing spaces
  - Empty strings
  - Unique values
  - Numerical ranges

- No major structural inconsistencies were observed during the initial assessment.

---

### 6. Important Observations
- Salary information is available in both textual and numerical formats.
- Experience information is available as both text and numerical ranges.
- Company ratings and review counts are available for a subset of companies.
- Job descriptions contain rich textual information suitable for advanced text analysis and skill extraction.
- The dataset contains sufficient diversity in companies, locations, salary ranges, experience levels, and required skills for meaningful analytics.

---

### 7. Project Potential
Based on the initial assessment, this dataset is suitable for developing:

- Interactive analytical dashboard
- Salary trend analysis
- Experience-wise job analysis
- Company insights
- Location-based hiring analysis
- Skill demand analysis
- Company rating analysis
- Job market trends
- Student Career Advisor
- SQL-powered filtering and querying
- Skill extraction and text analytics

---

### Conclusion
The dataset demonstrates good diversity, sufficient volume, and high analytical potential for developing a comprehensive job market analytics platform. Overall, the initial assessment indicates that the dataset is well-suited for exploratory data analysis, data preprocessing, visualization, dashboard development, and advanced analytical tasks.